### ISSUE TITLE
Improving label management

### ISSUE THREAD

**Issue by tcojean** [ginkgo-project/ginkgo#568]
We have a lot of labels and I believe we can all agree that we need all of these to properly classify PR/Issues, maybe we would even actually need more than what we currently have.
Nonetheless, I personally think we have so much that we might need a proper classification system. That is what I want to discuss here.

What I mean by that is that we could prefix labels by some system to categorize them, so that you can scroll through the labels by categories if needed. As an example we could have something like:
+ For all status qualification (WIP, Ready for review, ready to merge, maybe backlog would also make sense or some other ones), `st:WIP`, `st:READY-FOR-REVIEW`, or some abbreviation such as `st:RFR`. You get the idea. This way, typing `st` in the label search would allow you to easily switch between PR status.
+ For the modules, (core, HIP, ...) we could use a `re` for `regarding`, such as `re:core`, `re:hip`, ...
+ For qualifying the PR/issue, (enhancement, new feature, ...) we could use a `is` prefix, such as `is:new`, `is:enhancement`, etc.

What do you think about such a system? Do you think we need to improve label management to begin with? Maybe you would rather go into a totally different direction?

**Comment by upsj** [comment]
I really like the idea! One thing I am unsure about is in general how thorough we need to be with these labels:
* If we add a new feature, do we need to specify all executors that we implemented it on (which would usually be all of them)? Or is this only necessary when we make change to specific executors?
* The same question holds for core code - does any change to core justify such a label?
* What about testing? Do we add the label to any PR that contains tests, or only to ones that specifically make changes to testing?

Another category would be auxiliary tags like *Help Wanted*, *Affects Performance*. 

I am kind of torn on these questions. On the one hand, it is helpful to be able to enumerate all PRs that made changes to a certain part of Ginkgo, on the other hand, the large amount of labels can hide important ones.

And we should maybe think about issues as well. Many labels fit both, but I guess we need some issue-specific ones like *confirmed*, ...

**Comment by tcojean** [comment]
Yes, the issue you bring of label attribution is another one which I'm also unsure about, but definitely feels weird. On the one hand, it's nice to know every single PR which made a change say to HIP, but if anyway the PR adds a new algorithm, it will definitely also touch HIP.
Maybe if we have categories we can solve this issue with a few more labels? For example, in this case a `re:new` or `re:all` would state that this is a new algorithm and touches everything. Right now github doesn't support `OR` searches, but I hope they will at some point...

For testing, I'm fine with adding the testing label only when actually touching the testing framework in some way (extension, rewrite, ...). I believe any PR which has either a `new feature` or a `enhancement` should also implement tests anyway.

**Comment by pratikvn** [comment]
Uniformization is definitely a good idea. I actually renamed the labels a few days ago. One thing we might want to consider is the order the labels would appear next to the Issue/PR names. According to what I saw, the ordering is numerals first, then uppercase characters and then lower case. That is the reason I named all the status to upper case, so that they appear first in the list of labels as I believe this is something that should appear before the other qualifications.

If we add categories to those labels such as `st:`, maybe we can also make sure that the order in which the labels appear is also a reasonable one ? Maybe we can use:

`ST`: for the status, 
`reg:` for the regarding and so on..

**Comment by thoasm** [comment]
I also like the idea, especially if it allows us to sort them properly.
Here is an idea for 4 different prefixes in order:
1. `ST` for status (`WIP`, `ready for review`, ...), since I consider them the most important labels
2. `at` for attribute (or maybe `is` as you suggested) (`bug`, `confirmed`, `interface-breaking`, `high importance`)
3. maybe `mod` for module or `part` (`multigrid`, `preconditioner`, `solver`)
4. `rg` or `reg` for regarding (`core`, `hip`, ..., `testing`, `build`)

I think in terms of ordering, (1) and (4) should be the first and last respectively. We can surely add more prefixes, these are just the ones I came up with.

**Comment by tcojean** [comment]
I just did this with a few changes. Feel free to edit the labels again or propose something if you have some ideas.

+ I add a new `plat` for `platform` for `apple`, `windows` and `intel`.
+ I actually used the `mod` for actual Ginkgo modules, i.e. `core`, `cuda`, `openmp`, `hip`, `sycl` (future), `all`
+ I use `type` to designate `preconditioner`, `solver`, ... and all other ginkgo internal types/features
+ I use `regarding` for all the rest, e.g. `testing`, `build`, `ci`, ...
